web_src/js/features/repo-editor.js

```diff
@@ -120,18 +124,25 @@ export function initRepoEditor() {
         $(this)[0].setSelectionRange(0, 0);
       }
     }
-    parts = [];
-    $('.breadcrumb span.section').each(function () {
-      const element = $(this);
-      if (element.find('a').length) {
-        parts.push(element.find('a').text());
-      } else {
-        parts.push(element.text());
-      }
-    });
-    if ($(this).val()) parts.push($(this).val());
-    $('#tree_path').val(parts.join('/'));
-  }).trigger('keyup');
+
+    joinTreePath($(this));
+  });
+
+  $editFilename.on('keyup', function (e) {
+    const $section = $('.breadcrumb span.section');
+    const $divider = $('.breadcrumb div.divider');
+    let value;
+
+    if (e.keyCode === 8 && getCursorPosition($(this)) === 0 && $section.length > 0) {
+      value = $section.last().find('a').text();
+      $(this).val(value + $(this).val());
+      $(this)[0].setSelectionRange(value.length, value.length);
+      $section.last().remove();
+      $divider.last().remove();
+    }
+
+    joinTreePath($(this));
```

Comment:
One more last thought 😂 

Could this call be moved into the `if` above? This call is only needed when the path changes.

Reply:
> One more last thought 😂
> 
> Could this call be moved into the `if` above? This call is only needed when the path changes.

Sure that makes sense

Reply:
Oh, another last thought ... many `$()` could be also moved to out scope? Then we do not need to spend CPU time on every keyup/input event.

eg: 

```
    const $section = $('.breadcrumb span.section');
    const $divider = $('.breadcrumb div.divider');
    $editFilename.on('keyup', ... )
```

Reply:
> Oh, another last thought ... many `$()` could be also moved to out scope? Then we do not need to spend CPU time on every keyup/input event.

This will cause problem. Because the events should get the lastest elements' status(like `$section.length`). If they are intialized outside,  the elements will have the status on created.